LUCENE-10234: Change module prefix to org.apache.* (#487)
apache/lucene: lucene/CHANGES.txt

Improvements
---------------------

> * LUCENE-10234: Added Automatic-Module-Name to all JARs. This is the first step to enable full Java

Reviewer: I just browsed through the CHANGES and found there is `Build` section in the 9.0 changes.
https://github.com/apache/lucene/blob/749b744c0c2fcc5de709c3b48fa8ef802af010f4/lucene/CHANGES.txt#L473

Does it make sense to move this entry to there, what do you think?
Also, it seems to me LUCENE-9077 should appear at the top of the Build section, and LUCENE-8768 can be there too.

Reviewer: If you don't mind, can I edit the CHANGES.txt here ...?

Reviewer 2: You can add dawid's github as upstream and checkout his branch, edit it and push.

If you just want to change a bit of text, go into the diff view of the PR, scroll to the file (CHANGES.txt) and press on the three dots top-right and choose "Edit file".

Reviewer 2: But to come back to your original question: This is an "improvement" (we have module names now) and not a "build change" (yes it is a build change technically, but it does not improve our build system).

So I would leave it where it is.

Reviewer: Yes, I know I can edit that, I just wanted to ask the author's thoughts.

> So I would leave it where it is.

Ok, thanks.

After writing this suggestion, I noticed re-organizing CHANGES could complicate back-port; it could be better to untouch the file after cutting the release branches (unless it should be really fixed).

Improvements
---------------------

> * LUCENE-10234: Added Automatic-Module-Name to all JARs. This in a first step to enable full Java

Author: ```suggestion
* LUCENE-10234: Added Automatic-Module-Name to all JARs. This is the first step to enable full Java
```
